transformer_engine/common/dropout/dropout.cu

```diff
@@ -0,0 +1,284 @@
+/*************************************************************************
+ * Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include <curand.h>
+#include <curand_kernel.h>
+#include <curand_philox4x32_x.h>
```

Comment:
If we add cuRAND as a dependency, we may need to add logic to manually load `libcurand.so`, similar to how we load `libcublas.so`:
https://github.com/NVIDIA/TransformerEngine/blob/858755c0c3898d5bc971fcec031b26ff26e7e4fb/transformer_engine/common/__init__.py#L306-L307
This is because users might not have the CUDA Toolkit installed, but rather install CUDA libs with [Pip](https://pypi.org/project/nvidia-curand-cu12/).